### PR TITLE
XML2GFF: Add initial xml2gff tool

### PR DESCRIPTION
**Pull request by Nostritius**
This adds an initial xml2gff tool. I modified the gff3 writer, taken from xoreos, slightly, so it won't require glm

### CODE REVIEW

**Comment by DrMcCoy** [comment]
You're going C++11 mode now, with that range-based for? :P

Our Travis CI config is not yet adapted for that. But yes, I did say that I'm opening xoreos-tools up for C++11, so I guess it's time I put my money where my mouth is.

I'll see if I find time to change that during the week. :)

File: src/xml/gffcreator.cpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	const XMLNode &xmlRoot = xml.getRoot();
+
+	uint32 typeId;
+	Common::UString type = xmlRoot.getProperty("type");
+	for (size_t i = 0; i < 4; ++i) {
+		if (type.size() > i)
+			reinterpret_cast<char*>(&typeId)[3 - i] = *type.getPosition(i);
+		else
+			reinterpret_cast<char*>(&typeId)[3 - i] = ' ';
+	}
```
Comment (DrMcCoy) [inline src/xml/gffcreator.cpp]: That's...pretty icky. Can you use `MKTAG()` instead?

You don't even need the loop, you can just pad the string with 4 spaces beforehand. 

Like
```
const Common::UString type = xmlRoot.getProperty("type") + "    ";
const uint32 typeID = MKTAG(*type.getPosition(0), ...
```

(Theoretically, you could even use Common::UString::at() from a6f1c2bdb53785da394dc9dbdb146245eee72925)

File: src/xml/gff3creator.h [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+public:
+	GFF3Creator(uint32 id);
+
+	void create(const XML::XMLNode &root, Common::WriteStream &file);
+
+private:
+	void readStructContents(const XMLNode::Children &strctNodes, Aurora::GFF3WriterStructPtr strctPtr);
+	void readListContents(const XMLNode::Children &listNodes, Aurora::GFF3WriterListPtr listPtr);
+
+	Common::ScopedPtr<Aurora::GFF3Writer> _gff3;
```
Comment (DrMcCoy) [inline src/xml/gff3creator.h]: Hmm, can you push the state into the method parameters, and make `create()` static?

File: src/xml/gff3creator.cpp [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+			if (!strctNode->getChildren().empty()) {
+				for (const auto &child : strctNode->getChildren()) {
+					if (child->getName() != "string")
+						throw Common::Exception("GFF3Creator::readStructContents() Invalid LocString string");
+
+					uint32 language;
+					Common::parseString(child->getProperty("language"), language);
+
+					// TODO: Add gender if available.
```
Comment (DrMcCoy) [inline src/xml/gff3creator.cpp]: Can you fulfill this TODO first? I'd rather not have a tool available that accidentally removes information when doing a round-trip

File: src/rules.mk [hunk 4]
```diff
@@ -310,6 +310,19 @@ src_erf_LDADD = \
...
+    src/util.cpp \
+    $(EMPTY)
+src_xml2gff_LDADD = \
+    src/xml/libxml.la \
+    src/aurora/libaurora.la \
+    src/common/libcommon.la \
+    src/version/libversion.la \
+    $(LDADD) \
+    $(EMPTY)
+
```
Comment (DrMcCoy) [inline src/rules.mk]: Can you add the binary (both Unix and Windows) to the .gitignore file too, please?

(Also, a manpage would be great as well)

File: src/xml2gff.cpp [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+	using Common::CLI::ValAssigner;
+	using Common::CLI::makeEndArgs;
+	using Common::CLI::makeAssigners;
+	std::vector<Common::UString> args;
+	NoOption filesOpt(false, new ValGetter<std::vector<Common::UString> &>(args,
+	                                                                       "[input file] <output file>"));
+	Parser parser(argv[0], "XML to BioWare GFF converter",
+	              "If no input file is given, the input is read from stdin.\n\n"
+	              "Since different games use different SSF file version, specifying the\n"
+	              "game for which to create the SSF file is necessary.",
```
Comment (DrMcCoy) [inline src/xml2gff.cpp]: That's obviously wrong for a GFF tool :P

File: src/xml2gff.cpp [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+		inFile  = args[0];
+		outFile = args[1];
+	} else
+		outFile = args[0];
+
+	return true;
+}
+
+void createGFF(const Common::UString &inFile, const Common::UString &outFile) {
+	Common::WriteFile ssf(outFile);
```
Comment (DrMcCoy) [inline src/xml2gff.cpp]: Also not the right variable name here :P

**Comment by Nostritius** [comment]
The issues should now be fixed and I added a man page

**Comment by DrMcCoy** [comment]
Merged as 367563e...41aaf31, thanks! :)

**Pull request closed without merging**